### PR TITLE
Solved issue #573 (Configuration of label sizes for x,y axes and plot title)

### LINKED ISSUE #573: Plot x-axis, y-axis, title, and color bar label sizes should be configurable.
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/573) by Shawn Honomichl (Bitbucket: [Shawn](https://bitbucket.org/Shawn), GitHub: [Shawn](https://github.com/Shawn)).**

----------------------------------------

Plot x-axis, y-axis, title, and color bar label sizes should be configurable in the options.  During briefings when flight plan images are displayed, people often complain of having trouble seeing labels.

### PR DESCRIPTION
**Pull request by risehr**
Added Fontsize changing options in Sideview options for labels, ticklabels, and title of the sideview plot.

One option for changing size of ticklabels of x and both y axes.
Another option for changing x,y axes labels and plot title.

### CODE REVIEW

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
the amount of lines in the ui sideview_options.py looks a bit strange

**Comment by ReimarBauer** [comment]
sideview_image

**Comment by risehr** [comment]
I have not altered the ui_sideview_options.py manually. Infact, I have converted it from `pyuic5 -x ui_sideview_options.ui -o ui_sideview_options.py`. Also, I have designed the ui_sideview_options.ui file from the qt designer.

File: mslib/msui/mpl_qtwidget.py [hunk 1]
```diff
@@ -488,6 +488,8 @@ def __init__(self, model=None, settings=None, numlabels=None):
         self.settings_dict = {"vertical_extent": (1050, 180),
                               "vertical_axis": "pressure",
                               "secondary_axis": "no secondary axis",
+                              "plot_title_size": "default",
```
Comment (ReimarBauer) [inline mslib/msui/mpl_qtwidget.py:498]: we maybe should use here the config_loader and add the default value to the configuration. By this anyone can use his sizes from the mss_settings.json

https://github.com/Open-MSS/MSS/blob/develop/mslib/msui/__init__.py

We could add a dictionary here and start with 
```
topview = {"plot_title_size": 10,
                 "axes_label_size": 10}

sideview = {"plot_title_size": 10,
                 "axes_label_size": 10}

```

the config_loader can access this by 

topview_settings=config_loader(dataset="topview")

File: mslib/msui/qt5/ui_sideview_options.py [hunk 2]
```diff
@@ -269,3 +316,47 @@ def retranslateUi(self, SideViewOptionsDialog):
...
+        self.cbaxessize.setItemText(10, _translate("SideViewOptionsDialog", "22"))
+        self.cbaxessize.setItemText(11, _translate("SideViewOptionsDialog", "24"))
+        self.cbaxessize.setItemText(12, _translate("SideViewOptionsDialog", "26"))
+        self.cbaxessize.setItemText(13, _translate("SideViewOptionsDialog", "28"))
+        self.cbaxessize.setItemText(14, _translate("SideViewOptionsDialog", "30"))
+        self.cbaxessize.setItemText(15, _translate("SideViewOptionsDialog", "32"))
+        self.label_7.setText(_translate("SideViewOptionsDialog", "         Axes Label Size  "))
+
+
+if __name__ == "__main__":
```
Comment (ReimarBauer) [inline mslib/msui/qt5/ui_sideview_options.py:380]: on a converted ui file we don't have this usually. This may get added because of an option used?

```
 pyuic5 ui_sideview_options.ui > ../qt5/ui_sideview_options.py
```

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
needs to be done for topview too

**Comment by ReimarBauer** [comment]
> I have not altered the ui_sideview_options.py manually. Infact, I have converted it from `pyuic5 -x ui_sideview_options.ui -o ui_sideview_options.py`. Also, I have designed the ui_sideview_options.ui file from the qt designer.

sorry for my mistake, corrected already my comment

**Comment by risehr** [comment]
Okay, I will do for the topview also. And will also try to implement "the config loader part" that you said.

File: mslib/msui/qt5/ui_sideview_options.py (hunk 2 above)
Comment (risehr) [reply mslib/msui/qt5/ui_sideview_options.py:380]: > on a converted ui file we don't have this usually. This may get added because of an option used?
> 
> ```
>  pyuic5 ui_sideview_options.ui > ../qt5/ui_sideview_options.py
> ```

may be...i don't know exactly how that came.

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
the pyuic5 converter surprised me with that kind of output. But it is doing this.

Please can you look on setting up a default value in the MissionSupportSystemDefaultConfig and use this by the config_loader.

It works pretty well for sideview. We also need it for topview options.

The ui elements should be seperated from the Performance section into an own section.
Performance is used for aircraft specifications. e.g. weight, fuel.

A new section may be Font sizes?

File: mslib/msui/qt5/ui_sideview_options.py (hunk 2 above)
Comment (ReimarBauer) [reply mslib/msui/qt5/ui_sideview_options.py:380]: these `__main__` lines are never get into a test coverage but likly are counted as lines of the project.

File: mslib/msui/mpl_qtwidget.py (hunk 1 above)
Comment (risehr) [reply mslib/msui/mpl_qtwidget.py:498]: > we maybe should use here the config_loader and add the default value to the configuration. By this anyone can use his sizes from the mss_settings.json
> 
> https://github.com/Open-MSS/MSS/blob/develop/mslib/msui/__init__.py
> 
> We could add a dictionary here and start with
> 
> ```
> topview = {"plot_title_size": 10,
>                  "axes_label_size": 10}
> 
> sideview = {"plot_title_size": 10,
>                  "axes_label_size": 10}
> ```
> 
> the config_loader can access this by
> 
> topview_settings=config_loader(dataset="topview")

Yes. I will look into it if I can do it.

File: mslib/msui/qt5/ui_sideview_options.py (hunk 2 above)
Comment (risehr) [reply mslib/msui/qt5/ui_sideview_options.py:380]: > these `__main__` lines are never get into a test coverage but likly are counted as lines of the project.

Can/should we delete the `__main__`  lines manually?

**Comment by risehr** [comment]
> the pyuic5 converter surprised me with that kind of output. But is is doing this.
> 
> Please can you look on setting up a default value in the MissionSupportSystemDefaultConfig and use this by the config_loader.
> 
> It works pretty well for sideview. We also need it for topview options.
> 
> The ui elements should be seperated from the Performance section into an own section.
> Performance is used for aircraft specifications. e.g. weight, fuel.
> 
> A new section may be Font sizes?

Yes, I have thought about it to put it in a new section. But I wondered, changing fontsize is increasing the performance of the viewing of flighttrack by the user. That's why! Alright, I will put it in a new section named "Font sizes"

File: mslib/msui/qt5/ui_sideview_options.py (hunk 2 above)
Comment (ReimarBauer) [reply mslib/msui/qt5/ui_sideview_options.py:380]: eventually the command has different options on windows. maybe there is a supress option?

Comment (risehr) [reply mslib/msui/qt5/ui_sideview_options.py:380]: Have to study about it.

**Comment by risehr** [comment]
Hello @ReimarBauer I am still on this. Topview also got covered for size configuration. Right now, I am setting the MSSDefaultconfig. Just thought to update you on the work.

**Comment by risehr** [comment]
Added fontsize configurable options for topview also.
User can now put his default value for fontsize from mss_settings
Gave a separate space for fontsizes in topview and sideview options UI.

File: mslib/msui/mpl_qtwidget.py [hunk 3]
```diff
@@ -1269,13 +1319,17 @@ def draw_legend(self, img):
                                                xticks=[], yticks=[],
                                                label="ax2", zorder=0)
                 self.legax.patch.set_facecolor("None")
+                # self.legax.set_yticklabels("ax2", fontsize=32, minor=False)
```
Comment (ReimarBauer) [inline mslib/msui/mpl_qtwidget.py:1352]: if comments are not needed for explainations, remove them

on all places.
if it is something you want to log ;) use logging.debug()

File: mslib/msui/qt5/ui_sideview_options.py [hunk 4]
```diff
@@ -287,3 +340,48 @@ def retranslateUi(self, SideViewOptionsDialog):
         self.groupBox_4.setTitle(_translate("SideViewOptionsDialog", "Performance"))
         self.cbDrawCeiling.setText(_translate("SideViewOptionsDialog", "draw ceiling altitude"))
         self.btCeilingColour.setText(_translate("SideViewOptionsDialog", "colour"))
+        self.groupBox_5.setTitle(_translate("SideViewOptionsDialog", "Font Sizes"))
```
Comment (ReimarBauer) [inline mslib/msui/qt5/ui_sideview_options.py]: May be name this box title: "Plot Options". Hmm, somehow in topview these are all "Plot Options"

This will be also valid when we add there more features in the future.

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
recently there was a linearview module added. Please add this feature with an Option button there too.

see other comments

File: mslib/msui/mpl_qtwidget.py (hunk 3 above)
Comment (risehr) [reply mslib/msui/mpl_qtwidget.py:1322]: Oka

> if comments are not needed for explainations, remove them
> 
> on all places.
> if it is something you want to log ;) use logging.debug()

Okay. Noted.

Comment (risehr) [reply mslib/msui/mpl_qtwidget.py:1352]: > if comments are not needed for explainations, remove them
> 
> on all places.
> if it is something you want to log ;) use logging.debug()

Okay Noted.

File: mslib/msui/qt5/ui_sideview_options.py (hunk 4 above)
Comment (risehr) [reply mslib/msui/qt5/ui_sideview_options.py]: > May be name this box title: "Plot Options". Hmm, somehow in topview these are all "Plot Options"
> 
> This will be also valid when we add there more features in the future.

Okay. I will change this to "Plot Options."

**Comment by risehr** [comment]
> recently there was a linearview module added. Please add this feature with an Option button there too.
> 
> see other comments

Yeah, I have noticed that before but was not sure to add because there were no previous customizations about anything. But not to worry, I will add the options for this Linearview also.

File: mslib/msui/_tests/test_mpl_map.py [hunk 5]
```diff
@@ -41,6 +41,7 @@ def test_no_coastsegs(self):
         # On land
         self.map.ax.set_xlim([2, 3])
         self.map.ax.set_ylim([48, 49])
+        self.map._draw_auto_graticule()
```
Comment (ReimarBauer) [inline mslib/msui/_tests/test_mpl_map.py:44]: what makes this change?

Comment (ReimarBauer) [inline mslib/msui/_tests/test_mpl_map.py:44]: @risehr is this change needed for the test? If so you likly found a bug.

File: mslib/msui/mpl_map.py [hunk 6]
```diff
@@ -145,10 +144,8 @@ def __init__(self, identifier=None, CRS=None, BBOX_UNITS=None, PROJECT_NAME=None
                     self.crs_text = self.ax.figure.text(0, 0, self.crs)
 
         if self.appearance["draw_graticule"]:
-            try:
```
Comment (ReimarBauer) [inline mslib/msui/mpl_map.py:148]: @joernu76 please look on this change

**Review by ReimarBauer — CHANGES_REQUESTED** [review]
a few small questions added. 

used it with the 3 views and it is a good improvement :)

File: mslib/msui/mpl_map.py (hunk 6 above)
Comment (risehr) [reply mslib/msui/mpl_map.py:148]: @ReimarBauer Yes, the test was failing. Hence, it was needed. Both are related. Since, I removed `_draw_auto_graticule()` from def __init__() of mpl_map because this is already being called in __init__() of  MPLTopview Class of mpl_qtwidget(since it is necessary to draw graticule with the already changed fontsizes). And hence, since I removed this `_draw_auto_graticule()` from mpl_map.py, the test also needed to be changed otherwise the test would continue to fail.

Comment (ReimarBauer) [reply mslib/msui/mpl_map.py:148]: ok, the test is likly now on the"wrong" place, but we can move it later. 
We do avoid this when the test is using MPLTopview

**Review by ReimarBauer — APPROVED** [review]
